test/framework/framework.go

```diff
@@ -0,0 +1,537 @@
+	i.require.NoError(err)
+
+	if resp.StatusCode != http.StatusOK {
+		return "", fmt.Errorf("expected %v, but /status returned %v: %v", http.StatusOK, resp.Status, string(body))
+	}
+
+	return string(body), nil
+}
+
+// MarbleConfig contains the configuration for a Marble.
+type MarbleConfig struct {
+	coordinatorAddr string
+	marbleType      string
+	dnsNames        string
+	dataDir         string
+}
+
+// NewMarbleConfig creates a new MarbleConfig.
+func NewMarbleConfig(coordinatorAddr, marbleType, dnsNames string) MarbleConfig {
+	dataDir, err := os.MkdirTemp("", "")
+	if err != nil {
+		panic(err)
+	}
+	return MarbleConfig{
+		coordinatorAddr: coordinatorAddr,
+		marbleType:      marbleType,
+		dnsNames:        dnsNames,
+		dataDir:         dataDir,
+	}
+}
+
+// Cleanup removes the data directory of the Marble.
+func (c MarbleConfig) Cleanup() {
+	if err := os.RemoveAll(c.dataDir); err != nil {
+		panic(err)
+	}
+}
+
+// GetMarbleCmd returns the command to start a Marble.
+func (i IntegrationTest) GetMarbleCmd(cfg MarbleConfig) *exec.Cmd {
+	var cmd *exec.Cmd
+	if i.NoEnclave {
+		cmd = exec.Command(filepath.Join(i.BuildDir, "marble-test-noenclave"))
+	} else {
+		cmd = exec.Command("erthost", filepath.Join(i.BuildDir, "marble-test-enclave.signed"))
+	}
+	cmd.SysProcAttr = &syscall.SysProcAttr{Pdeathsig: syscall.SIGKILL} // kill if parent dies
+	uuidFile := filepath.Join(cfg.dataDir, "uuid")
+	cmd.Env = []string{
+		MakeEnv(mconfig.CoordinatorAddr, cfg.coordinatorAddr),
+		MakeEnv(mconfig.Type, cfg.marbleType),
+		MakeEnv(mconfig.DNSNames, cfg.dnsNames),
+		MakeEnv(mconfig.UUIDFile, uuidFile),
+		MakeEnv("EDG_TEST_ADDR", i.MarbleTestAddr),
+		i.SimulationFlag,
+	}
+	return cmd
+}
+
+// StartMarbleServer starts a Server Marble.
+func (i IntegrationTest) StartMarbleServer(cfg MarbleConfig) *os.Process {
+	cmd := i.GetMarbleCmd(cfg)
+	output := i.StartCommand(cmd)
+
+	log.Println("Waiting for server...")
+	timeout := time.Second * 5
+	for {
+		time.Sleep(100 * time.Millisecond)
+		select {
+		case out := <-output:
+			// process died
+			log.Println(out)
+			return nil
+		default:
+		}
+		conn, err := net.DialTimeout("tcp", i.MarbleTestAddr, timeout)
+		if err == nil {
+			conn.Close()
+			log.Println("Server started")
+			return cmd.Process
+		}
+	}
+}
+
+// StartMarbleClient starts a Client Marble.
+func (i IntegrationTest) StartMarbleClient(cfg MarbleConfig) bool {
+	out, err := i.GetMarbleCmd(cfg).CombinedOutput()
+	if err == nil {
+		return true
+	}
+
+	if _, ok := err.(*exec.ExitError); ok {
+		return false
+	}
+
+	panic(err.Error() + "\n" + string(out))
+}
+
+// TriggerRecovery starts a Coordinator, sets the manifest, and triggers a recovery.
+func (i IntegrationTest) TriggerRecovery(manifest manifest.Manifest) ([]byte, *os.Process, *os.Process, CoordinatorConfig, MarbleConfig, string) {
+	// start Coordinator
+	log.Println("Starting a coordinator enclave")
+	cfg := NewCoordinatorConfig()
+	coordinatorProc := i.StartCoordinator(cfg)
```

Comment:
Why is this starting a new Coordiantor, and all other functions assume it's already running?

Reply:
(Please note that this PR only changes the existing code to run as methods of `IntegrationTest`)

I guess this is mostly comes down to implementation decisions when the recover test was created.
Imo the functions `TestRecoveryRestoreKey`, and `TestRecoveryReset` could be changed to run some of the tasks from the `TriggerRecovery` method

Reply:
Do you plan to do further refactoring in separate PRs?

Reply:
This PR was intended to provide us with an easier way to add new integration tests and adapt existing ones in the future.
Might be a good idea to refactor the recover tests in this PR

Reply:
Refactored the code so the test functions are responsible for starting the coordinator